### DISCUSSION TITLE
Heatmap with ColorRamp instead of density coloring

### DISCUSSION THREAD

**Discussion by valoricDe** [maplibre/maplibre-gl-js#1822] [Ideas]
Hi there,

I would like to achieve something like on this site: https://www.researchgate.net/figure/Color-shaded-relief-map-of-the-horizontal-gradient-derived-from-the-simple-Bouguer_fig5_258464984

A colored gradient based on some data points I have. Currently I can only color based on some weight of the heatmap.

I tried to define heatmap color based on my input data. But it seems to depend on heatmap-density solely

```
            'heatmap-color': [
              'interpolate',
              ['linear'],
              [
                '*',
                ['get', 'heatmap-density'],
                ['get', 'etaDeviationInSeconds']
              ],
              -2000,
              'rgba(0,73,194,0.85)',
              0,
              'rgba(76,178,4,0.5)',
              1,
              'rgb(178,24,43)',
              2000,
              'rgb(178,24,43)'
            ],
            ```
            
Error: layers.eta-heatmap-lyr.paint.heatmap-color: data expressions not supported

How difficult this would be, what do you think?

**Comment by mattbbc** [comment]
This is similar to a requirement I have.

I have a map that uses aggregation data from Elasticsearch to plot points, each point being one of the aggregation buckets that has some metadata regarding the number of documents located in that point. So each point has a 'count' value.

Is it possible to create a heatmap layer that uses this count value instead of the density of pins/points to determine the colours of the heatmap? Is this a feature that could be implemented - we have too high a point count to make it feasible to plot every point on the map without affecting performance.

**Reply by HarelM** [reply]
You can create a circle with radius based on the number of documents, have it a bit transparent.
IDK...
I'm having a hard time imagining what you would like to achieve, so knowing if it's possible or not is problematic...

**Comment by HarelM** [comment]
I think you can create a custom layer and implement that in it, or do a server side gradient raster tiles.
IDK, a visual mock up would help...
Also, maybe three slack channel is a better place to find people who are looking for the same solutions...

**Comment by cscetbon** [comment]
hey @mattbbc, how did you en up doing it ?

**Reply by mattbbc** [reply]
Sadly I wasn't able to figure out a way of creating a heat map layer based on pin attributes (count in my case).

**Reply by cscetbon** [reply]
@mattbbc I'm looking at https://github.com/Rylern/InterpolateHeatmapLayer for what it's worth

**Reply by mattbbc** [reply]
Thanks for the tip 🙂

**Comment by cscetbon** [comment]
@valoricDe what did you end up doing ? I'm facing the same issue and  https://github.com/Rylern/InterpolateHeatmapLayer doesn't give me the visual I want. I'm really surprised this is not something that has been implemented for a long time. I want to draw a heatmap made up of values that go from good (green) to bad (red) but don't want the zoom to aggregate everything into the same color because it uses the density.
